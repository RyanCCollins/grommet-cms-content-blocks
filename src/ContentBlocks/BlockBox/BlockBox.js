/* @flow */
import React from 'react';
import Box from 'grommet/components/Box';
import Markdown from 'grommet/components/Markdown';

export default function BlockBox(props: {
  content: ?string,
  colorIndex: ?string,
}) {
  const { content, colorIndex } = props;
  return (
    <Box
      align="center"
      justify="center"
      size={{ height: 'small' }}
      colorIndex={colorIndex || 'light-1'}
    >
      <Box pad="small">
        <Markdown
          content={content || ''}
          components={{
            'p': { 'props':  { size: 'large', margin: 'small' } },
            'h2': { 'props':  { strong: true } }
          }}
        />
      </Box>
    </Box>
  );
}
