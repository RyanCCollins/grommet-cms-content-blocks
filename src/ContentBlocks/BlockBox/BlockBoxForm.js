/* @flow */
import React from 'react';
import Box from 'grommet/components/Box';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import FormField from 'grommet/components/FormField';
import Select from 'grommet/components/Select';
import Button from 'grommet/components/Button';
import Footer from 'grommet/components/Footer';
import colorOptions from './internal/colorIndexes';

type ErrorType = string;

type BlockColorSwatchFormState = {
  colorIndexInput: ?string,
  contentInput: ?string,
  errors: ?{
    colorIndexInput: ?ErrorType,
    contentInput: ?ErrorType
  }
};

type BlockColorSwatchFormProps = {
  onSubmit?: Function,
  colorIndex: ?string,
  content: ?string
};

export default class BlockColorSwatchForm extends React.Component {
  state: BlockColorSwatchFormState;
  props: BlockColorSwatchFormProps;
  constructor(props: BlockColorSwatchFormProps) {
    super(props);
    (this:any)._onChange = this._onChange.bind(this);
    (this:any)._onSubmit = this._onSubmit.bind(this);
    (this:any)._formIsValid = this._formIsValid.bind(this);
    let colorIndexInput = '';
    let contentInput = '';
    const { colorIndex, content } = props;
    if (colorIndex) {
      colorIndexInput = colorIndex;
    }
    if (content) {
      contentInput = content;
    }
    this.state = {
      errors: null,
      colorIndexInput,
      contentInput
    };
  }
  
  _onChange({ target, option }: any) {
    if (option) {
      this.setState({
        [`${target.id}`]: option,
        errors: this.state.colorIndexInput
          ? null
          : this.state.errors
      });
    } else {
      this.setState({
        [`${target.id}`]: target.value,
        errors: this.state.contentInput
          ? null
          : this.state.errors
      });
    }
  }

  _onSubmit(event: any) {
    event.preventDefault();
    const { colorIndexInput, contentInput } = this.state;
    if (this._formIsValid() && this.props.onSubmit) {
      this.props.onSubmit({
        colorIndex: colorIndexInput,
        content: contentInput
      });
    } else {
      const errors = {
        colorIndexInput: colorIndexInput && colorIndexInput.length > 0
          ? '' : 'Please enter a valid color index',
        contentInput: contentInput && contentInput.length > 0
          ? '' : 'Please enter some content for the box'
      };
      this.setState({
        errors
      });
    }
  }

  _formIsValid() {
    const { colorIndexInput, contentInput } = this.state;
    if (colorIndexInput && contentInput) {
      return colorIndexInput.length > 0 && contentInput.length > 0;
    }
    return false;
  }

  render() {
    const {
      colorIndexInput,
      contentInput,
      errors
    } = this.state;
    return (
      <Box colorIndex="light-2" pad="medium">
        <Form>
          <FormFields>
            <fieldset>
              <FormField
                label="Content"
                htmlFor="contentInput"
                error={errors && errors.contentInput ? errors.contentInput : ''}
              >
                <textarea
                  id="contentInput"
                  name="contentInput"
                  type="text"
                  value={contentInput}
                  onChange={this._onChange}
                  rows="10"
                />
              </FormField>
              <FormField
                label="Color Index"
                htmlFor="colorIndexInput"
                help="The color index for the box.  See: https://grommet.github.io/docs/color"
                error={errors && errors.colorIndexInput ? errors.colorIndexInput : ''}
              >
                <Select
                  onChange={this._onChange}
                  value={colorIndexInput || ''}
                  options={colorOptions}
                  name="colorIndexInput"
                  id="colorIndexInput"
                />
              </FormField>
            </fieldset>
          </FormFields>
          <Footer pad="medium">
            <Button
              onClick={this._onSubmit}
              type="submit"
              label="Done"
            />
          </Footer>
        </Form>
      </Box>
    );
  }
}

