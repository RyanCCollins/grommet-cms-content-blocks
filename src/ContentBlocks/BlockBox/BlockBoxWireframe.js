import React from 'react';
import Box from 'grommet/components/Box';

export default function BlockBoxWireframe() {
  return (
    <Box pad={{ between: 'small' }}>
      <Box pad="medium" colorIndex="accent-3" />
    </Box>
  );
}
