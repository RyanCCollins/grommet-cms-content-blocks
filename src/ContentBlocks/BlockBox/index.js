export BlockBox from './BlockBox';
export BlockBoxForm from './BlockBoxForm';
export BlockBoxWireframe from './BlockBoxWireframe';
