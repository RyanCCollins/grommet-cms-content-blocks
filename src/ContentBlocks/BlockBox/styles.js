import { css } from 'styled-components';

export default (borderColor: ?string) => {
  if (borderColor !== '') {
    return css`
      border-bottom: 9px solid ${borderColor};
    `;
  }
  return css``;
};
