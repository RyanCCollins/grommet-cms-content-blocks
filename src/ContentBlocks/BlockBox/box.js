import styled from 'styled-components';
import Box from 'grommet/components/Box';
import styles from './styles';

export default styled(Box)`
  ${props => styles(props.borderColor || '')}
`;
