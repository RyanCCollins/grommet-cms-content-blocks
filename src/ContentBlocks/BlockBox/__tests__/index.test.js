import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import BlockBox from '../BlockBox';

const props = {
  content: 'foo',
  align: 'start',
  colorIndex: 'brand',
};

describe('<BlockBox />', () => {
  describe('when shallow rendered', () => {
    it('should create a snapshot with default props', () => {
      const wrapper = shallow(
        <BlockBox />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should create a snapshot with custom props', () => {
      const wrapper = shallow(
        <BlockBox {...props} />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should create a snapshot with borderColor prop', () => {
      const final = {
        ...props,
        borderColor: '#fff',
      };
      const wrapper = shallow(
        <BlockBox {...final} />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
