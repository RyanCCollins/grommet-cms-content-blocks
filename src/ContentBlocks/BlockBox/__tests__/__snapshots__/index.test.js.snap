// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockBox /> when shallow rendered should create a snapshot with borderColor prop 1`] = `
<Styled(Box)
  align="start"
  borderColor="#fff"
  colorIndex="brand"
  justify="center"
  size={
    Object {
      "height": "small",
    }
  }
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="small"
    responsive={true}
    tag="div"
  >
    <BlockParagraph
      align="start"
      content="foo"
    />
  </Box>
</Styled(Box)>
`;

exports[`<BlockBox /> when shallow rendered should create a snapshot with custom props 1`] = `
<Styled(Box)
  align="start"
  colorIndex="brand"
  justify="center"
  size={
    Object {
      "height": "small",
    }
  }
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="small"
    responsive={true}
    tag="div"
  >
    <BlockParagraph
      align="start"
      content="foo"
    />
  </Box>
</Styled(Box)>
`;

exports[`<BlockBox /> when shallow rendered should create a snapshot with default props 1`] = `
<Styled(Box)
  align="center"
  colorIndex="light-1"
  justify="center"
  size={
    Object {
      "height": "small",
    }
  }
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="small"
    responsive={true}
    tag="div"
  >
    <BlockParagraph
      align="center"
      content=""
    />
  </Box>
</Styled(Box)>
`;
